packages/babel-traverse/src/path/inference/inferers.js

```diff
@@ -137,7 +137,13 @@ function Func() {
   return t.genericTypeAnnotation(t.identifier("Function"));
 }
 
-export { Func as Function, Func as Class };
+export {
+  Func as FunctionExpression,
```

Comment:
And IIRC this was actually _broken_ the old way - i.e. whoever wrote this was clearly expecting some code to expand `Function` and `Class` a la visitor method names, and it wasn't happening. (But not tested so it only came up when I tackled coverage)